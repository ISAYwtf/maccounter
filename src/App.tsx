import React from 'react'
import './App.css'
import Header from "./components/Header/Header.tsx";
import Sidebar from "./components/Sidebar/Sidebar";

const App = () =>
    <div className="app" data-mode={"light"}>
        <Header />
        <section className={"app-wrap"}>
            {/*<div className={"sidebar"}>*/}
                <Sidebar />
            {/*</div>*/}
            <main>
            </main>
        </section>
    </div>

export default App
