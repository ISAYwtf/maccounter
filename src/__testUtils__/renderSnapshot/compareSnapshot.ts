import { render } from '@testing-library/react'
import React, { ReactElement } from 'react'

export const compareSnapshot = (component: ReactElement, wrapper?: React.FC) => {
    const { asFragment } = render(component, { wrapper })
    expect(asFragment()).toMatchSnapshot()
}
