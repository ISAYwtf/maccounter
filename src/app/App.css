/* Свойства для темных и светлых режимов */

.app[data-theme="light"] {
    --color-bg-body: #F5F5F5;
    --color-bg-component: white;
    --color-txt: #313131;
    --color-main: #0E91EE;
    --color-addition-access: #00F083;
    --color-addition-warn: #FF9000;
    --color-addition-err: #FF4500;
    --color-disable: #E5E5E5;
    --color-side: #AEAEAE;
    --color-overlay: rgba(255, 255, 255, 0.7);
    --color-shadow: rgba(187, 187, 187, 0.25);
    --shadow-component: 0 10px 20px var(--color-shadow);
}

.app {
    color: var(--color-txt);
    background-color: var(--color-bg-body);
    width: 100vw;
    height: 100vh;
    overflow-x: hidden;
    display: flex;
    flex-direction: column;
}

/* Общие свойства */

.app-wrap {
    padding: 30px;
}

span {
    color: var(--color-main);
}

.txt--orange {
    color: var(--color-addition-warn);
}

.txt--logo {
    font-size: var(--size-font-logo);
}

.txt--title-small {
    font-size: var(--size-font-title-small);
}

.component {
    box-shadow: var(--shadow-component);
    border-radius: var(--size-border-rad-component);
    padding: var(--size-padding-component);
    background-color: var(--color-bg-component);
    height: fit-content;
}

/* Локальные свойства */

.app section {
    display: flex;
    padding-top: 20px;
    padding-bottom: 0;
    flex-grow: 1;
}
