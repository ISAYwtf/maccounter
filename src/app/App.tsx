import React, { FC } from 'react'
import './App.css'
import Header from '@components/Header/Header'
import Sidebar from '@components/Sidebar/Sidebar'
import { Provider } from 'react-redux'
import store from '@store/redux-store'
import { BrowserRouter } from 'react-router-dom'
import Modal from '@components/Modal/Modal'
import ContentPage from '@pages/ContentPage/ContentPage'

const App: FC = () => (
    <BrowserRouter>
        <Provider store={store}>
            <div className="app" data-theme="light">
                <Header />
                <section className="app-wrap">
                    <Sidebar />
                    <ContentPage />
                </section>
                <Modal />
            </div>
        </Provider>
    </BrowserRouter>
)

export default App
