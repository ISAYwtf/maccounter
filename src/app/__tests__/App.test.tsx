import React from 'react'
import { compareSnapshot } from '@testUtils/renderSnapshot'
import App from 'app/App'

describe('App', () => {
    const renderComponent = <App />

    // eslint-disable-next-line
    it('render App', () => compareSnapshot(renderComponent))
})
