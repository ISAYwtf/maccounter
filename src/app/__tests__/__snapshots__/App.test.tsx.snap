// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render App 1`] = `
<DocumentFragment>
  <div
    class="app"
    data-mode="light"
  >
    <header
      class="header app-wrap"
    >
      <div
        class="headerLeft"
      >
        <div
          class="appLogo"
        >
          <div
            class="txt--logo"
          >
            <span
              class="txt--orange"
            >
              m.
            </span>
            <span>
              accounter
            </span>
          </div>
        </div>
        <div>
          <div
            class="txt--title-small"
          >
            <p>
              Good morning, 
              <span>
                User
              </span>
              !
            </p>
          </div>
        </div>
      </div>
      <div
        class="switcher"
      >
        <p>
          Light mode
        </p>
        <label
          class="switch"
        >
          <input
            type="checkbox"
          />
          <span
            class="slider"
          />
        </label>
      </div>
    </header>
    <section
      class="app-wrap"
    >
      <div
        class="sidebar component"
        role="navigation"
        style="top: 384px; width: 64px;"
      >
        <a
          class="sidebarItem"
          href="/profile"
          role="link"
        >
          <span
            class="sidebarProfile"
          >
            IA
          </span>
          <div
            class="prompt"
          >
            <p>
              Profile
            </p>
          </div>
        </a>
        <a
          class="sidebarItem"
          href="/home"
          role="link"
        >
          <svg>
            home.svg
          </svg>
          <div
            class="prompt"
          >
            <p>
              Home
            </p>
          </div>
        </a>
        <a
          class="sidebarItem"
          href="/accountancy"
          role="link"
        >
          <svg>
            accountancy.svg
          </svg>
          <div
            class="prompt"
          >
            <p>
              Accountancy
            </p>
          </div>
        </a>
        <a
          class="sidebarItem"
          href="/statistics"
          role="link"
        >
          <svg>
            statistics.svg
          </svg>
          <div
            class="prompt"
          >
            <p>
              Statistics
            </p>
          </div>
        </a>
        <a
          class="sidebarItem"
          href="/settings"
          role="link"
        >
          <svg>
            settings.svg
          </svg>
          <div
            class="prompt"
          >
            <p>
              Settings
            </p>
          </div>
        </a>
        <a
          class="sidebarItem"
          href="/add-operation"
          role="link"
        >
          <svg>
            add.svg
          </svg>
          <div
            class="prompt"
          >
            <p>
              Add operation
            </p>
          </div>
        </a>
      </div>
      <main
        class="container"
      >
        <div>
          <div
            class="component"
          >
            <div
              class="txt--title-small title"
            >
              <p>
                Last expenses
              </p>
              <svg>
                add.svg
              </svg>
            </div>
            <div
              class="rows"
            >
              <div
                class="row"
              >
                <div
                  class="column card"
                >
                  <p
                    class="cardTitle"
                  >
                    Mastercard
                  </p>
                  <p
                    class="cardNumber"
                  >
                    ****9474
                  </p>
                </div>
                <div
                  class="column sum"
                >
                  <p>
                    3,000 ₽
                  </p>
                </div>
                <div
                  class="column dots"
                >
                  <svg>
                    dots.svg
                  </svg>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="column card"
                >
                  <p
                    class="cardTitle"
                  >
                    Mastercard
                  </p>
                  <p
                    class="cardNumber"
                  >
                    ****9474
                  </p>
                </div>
                <div
                  class="column sum"
                >
                  <p>
                    3,000 ₽
                  </p>
                </div>
                <div
                  class="column dots"
                >
                  <svg>
                    dots.svg
                  </svg>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="column card"
                >
                  <p
                    class="cardTitle"
                  >
                    Mastercard
                  </p>
                  <p
                    class="cardNumber"
                  >
                    ****9474
                  </p>
                </div>
                <div
                  class="column sum"
                >
                  <p>
                    17,000 ₽
                  </p>
                </div>
                <div
                  class="column dots"
                >
                  <svg>
                    dots.svg
                  </svg>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="column card"
                >
                  <p
                    class="cardTitle"
                  >
                    Mastercard
                  </p>
                  <p
                    class="cardNumber"
                  >
                    ****9474
                  </p>
                </div>
                <div
                  class="column sum"
                >
                  <p>
                    1,150,000 ₽
                  </p>
                </div>
                <div
                  class="column dots"
                >
                  <svg>
                    dots.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </div>
</DocumentFragment>
`;
